docs/src/entry.jsx

```diff
@@ -0,0 +1,327 @@
+      const newSelectedLayout = (localStorage.getItem("aw2-settings-key-layout")) || "us";
+      if (newSelectedLayout !== selectedLayout) {
+        selectedLayout = newSelectedLayout;
+        updateKeymaps();
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    setTimeout(checkSettingsFromLocalStorage, 1000);
+  }
+
+  function saveKeyMap() {
+    localStorage.setItem("aw2-settings-key-mapping", JSON.stringify(keyMap));
+  }
+
+  function saveSelectedLayout() {
+    localStorage.setItem("aw2-settings-key-layout", selectedLayout);
+  }
+
+  function loadSpecializations() {
+    const specializations = document.querySelectorAll("[data-armory-embed=\"specializations\"]");
+    for (const specialization of specializations) {
+      const traitlines = [];
+      for (const line of (specialization.getAttribute("data-armory-ids") || "").split(",")) {
+        const id = parseInt(line, 10);
+        const defaultSelected = (specialization.getAttribute(`data-armory-${id}-traits`) || "").split(",").map((t) => parseInt(t, 10));
+        traitlines.push(<TraitLine id={id} defaultSelected={defaultSelected}/>);
+      }
+      ReactDOM.render(<APILanguageProvider value="en"><div>{traitlines}</div></APILanguageProvider>, specialization);
+    }
+  }
+  loadSpecializations();
+  function loadItems() {
+    const items = document.querySelectorAll("[data-aw2-item]");
+    for (const item of items) {
+      const id = parseInt(item.getAttribute("data-aw2-item"), 10);
+      ReactDOM.render(<APILanguageProvider value="en"><Item id={id} inline={true} /></APILanguageProvider>, item);
+    }
+  }
+  loadItems();
+
+  function loadSkillLoadout() {
+    const loadouts = Array.from(document.querySelectorAll("[data-armory-embed=\"skills\"][data-armory-ids]"));
+    loadouts.forEach((loadout) => {
+      const skillLoadout = [];
+      let i = 0;
+      for (const line of (loadout.getAttribute("data-armory-ids") || "").split(",")) {
+        const id = parseInt(line, 10);
+        const slotId = `${(i + 6) % 10}`;
+        skillLoadout.push(<div key={`slot-${slotId}`} className="aw2-skill-loadout-slot" data-aw2-key={slotId}>
+          <Skill id={id} disableText={true} style={{fontSize: "64px"}}/>
+          <div className="aw2-show-key">{localKey(keyFromDefault(slotId))}</div>
+        </div>);
+        i++;
+      }
+
+      ReactDOM.render(<APILanguageProvider value="en"><div className="aw2-skill-loadout" >{skillLoadout}</div></APILanguageProvider>, loadout);
+
+    });
+  }
+  loadSkillLoadout();
+
+  function loadSkills() {
+    const aw2Skills = Array.from(document.querySelectorAll("[data-aw2-skill]"));
+    aw2Skills.forEach((key) => {
+      if (!key.classList.contains("armory-inline")) {
+        const skillId = key.getAttribute("data-aw2-skill");
+        key.classList.add("armory-inline");
+        const armoryElement = document.createElement("span");
+        key.appendChild(armoryElement);
+        if (skillFallback[skillId]) {
```

Comment:
This is not a fallback. This is an override. If Anet does update the API to include the skill, we'll still use our fallback data. Either change the logic to reflect that, or the name to make it clear. I'd prefer an actual fallback, but name change is relatively easier.

Reply:
I'm now providing a real fallback. For a better solution we should add that functionality to gw2-ui

Reply:
Solution is there: https://github.com/discretize/discretize-ui/blob/main/gw2-ui/src/gw2api/overrides/data/skills.ts